_posts/2018-11-16-grpc-dotnet-build.md

```diff
@@ -0,0 +1,175 @@
+---
+layout: post
+title: gRPC Meets .NET SDK And Visual Studio
+published: true
+permalink: blog/grpc-dotnet-build
+author: Kirill 'kkm' Katsnelson
+author-link: https://github.com/kkm000
+company: SmartAction
+company-link: https://www.smartaction.ai
+---
+As part of Microsoft's move towards its cross-platform .NET offering, they have
+greatly simplified the project file format, and allowed a tight integration of
+third-party code generators with .NET projects. We are listening, and now proud
+to introduce integrated compilation of Protocol Buffer and gRPC service
+`.proto` files in .NET C# projects starting with the version 1.17 of the
+Grpc.Tools NuGet package, now available from Nuget.org.
+
+<!--more-->
+
+## A Walkthrough
+
+Now you can include `.proto` files along with C# source files into `.csproj`
+project files. In this blog post, we'll walk through the simplest and probably
+the most common scenario of creating a library from `.proto` files using the
+cross-platform `dotnet` command. We will implement essentially a clone of the
+`Greeter` library, shared by client and server projects in the [C# `Helloworld`
+example directory
+](https://github.com/grpc/grpc/tree/master/examples/csharp/Helloworld/Greeter).
+
+Start by creating a new library project.
+
+```
+kkm@yupana:~/work$ dotnet new classlib -o MyGreeter
+The template "Class library" was created successfully.
+
+kkm@yupana:~/work$ cd MyGreeter
```

Comment:
perhaps you want to omit the username and hostname in the snippets?

Reply:
Done. Certainly, this was just extra noise.